File: clients/python/tests/test_model.py
from pathlib import Path

import tinystan

STAN_FOLDER = Path(__file__).parent.parent.parent.parent / "test_models"


def test_model_loads():
    model = tinystan.Model(STAN_FOLDER / "bernoulli" / "bernoulli_model.so")
    assert model is not None


def test_api_version():
    model = tinystan.Model(STAN_FOLDER / "bernoulli" / "bernoulli_model.so")
    assert model.api_version() == tinystan.__version.__version_info__


def test_stan_version():
    model = tinystan.Model(STAN_FOLDER / "bernoulli" / "bernoulli_model.so")
    stan_version = model.stan_version()
    assert stan_version[0] == 2
    assert stan_version[1] >= 34
    assert stan_version[2] >= 0

File: clients/R/tests/testthat/test_model.R
bernoulli_file <- file.path(stan_folder, "bernoulli", "bernoulli.stan")
test_that("model loads", {
  suppressWarnings(model <- tinystan_model(bernoulli_file))
  expect_true(!is.null(model))
})

test_that("api_version is correct", {
  expect_equal(api_version(bernoulli_model), current_version_list)
})

test_that("stan version is valid", {
  stan_version <- stan_version(bernoulli_model)
  expect_equal(stan_version$major, 2)
  expect_gte(stan_version$minor, 34)
  expect_gte(stan_version$patch, 0)
})
